Solutions/Azure Cloud NGFW by Palo Alto Networks/Hunting Queries/CloudNGFW-HighRiskPorts.yaml

```diff
@@ -0,0 +1,112 @@
+  1434,"udp","Data Access/Mgmt","MS SQL Monitor Port",
+  1900,"udp","Hacker Recon/3rd Party Attacks","Simple Service Discovery Protocol, unencrypted",
+  2049,"tcp","Unsecure","Network File System",
+  2049,"udp","Unsecure","Network File System",
+  2301,"tcp","Hacker Recon","Compaq Management Service, no recent incidents",
+  2381,"tcp","Management","Compaq Management Service, no recent incidents",
+  3268,"tcp","Hacker Recon","Microsoft Global Catalog LDAP",
+  3306,"tcp","Data Access/Mgmt","MySQL Database Management Port",
+  3389,"tcp","Management/3rd Party Attacks","RDP, Common brute force attack port",
+  3389,"udp","Management/3rd Party Attacks","RDP, Common brute force attack port",
+  4333,"tcp","Data Access/Mgmt","MSql",
+  5353,"udp","3rd Party Attacks","mDNS",
+  5432,"tcp","Data Access/Mgmt","PostgresSQL Database Management",
+  5800,"tcp","Management","VNC Remote Frame Buffer over HTTP",
+  5900,"tcp","Management","VNC Remote Frame Buffer over HTTP",
+  5985,"tcp","Management","Windows Powershell",
+  5986,"tcp","Management","Windows Powershell",
+  6379,"tcp","Data Access/Mgmt","Redis",
+  7000,"tcp","Data Access/Mgmt","Cassandra",
+  7001,"tcp","Data Access/Mgmt","Cassandra",
+  7199,"tcp","Data Access/Mgmt","Cassandra",
+  9042,"tcp","Data Access/Mgmt","Cassandra",
+  9160,"tcp","Data Access/Mgmt","Cassandra",
+  9200,"tcp","Data Access/Mgmt","Elastic Search",
+  9300,"tcp","Data Access/Mgmt","Elastic Search",
+  9987,"udp","3rd Party Attack","DSM/SCM Target Interface",
+  11211,"udp","Unencrypted","Memcached",
+  16379,"tcp","Data Access/Mgmt","Redis",
+  26379,"tcp","Data Access/Mgmt","Redis",
+  27017,"tcp","Data Access/Mgmt","MongoDB",
+  ];
+  HighRiskPorts
+  | join kind=inner (
+    fluentbit_CL
+    | where FirewallName_s == "palocloudfw" and ident_s == "TRAFFIC" and Message.action != "deny"
+    | extend DestinationIP = tostring(parse_json(Message).dst_ip), DestinationPort = toint(parse_json(Message).dport), Protocol = tostring(parse_json(Message).proto), SourceIP = tostring(parse_json(Message).src_ip)
+    | where isnotempty(DestinationIP) and isnotempty(SourceIP)
+    //Remove private IP communication from DestinationIP
+    | extend result = ipv4_is_private(DestinationIP) 
+    | where result == 0
+    | summarize
+        Count = count(),
+        StartTime = min(TimeGenerated),
+        EndTime = max(TimeGenerated)
+        by 
+        FirewallName_s,
+        SourceIP,
+        DestinationIP,
+        DestinationPort,
+        Protocol
+  ) on $left.Port == $right.DestinationPort and $left.Protocol == $right.Protocol
+  | project-away Protocol1, Port
+  | order by FirewallName_s asc, SourceIP asc, DestinationIP asc, DestinationPort asc
+  | extend timestamp = StartTime, IPCustomEntity = SourceIP
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: SourceIP
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: DestinationIP
```

Comment:
Please add "version" for all the hunting queries

Reply:
Fixed now. Thanks.